include/alpaka/vec/Vec.hpp

```diff
@@ -555,6 +555,66 @@ namespace alpaka
             TSize m_data[TDim::value == 0u ? 1u : TDim::value];
         };
 
+        //-----------------------------------------------------------------------------
+        //! This is a conveniance method to have a out-of-class factory method even though the out-of-class version is not supported by all compilers.
+        //! This has the draw-back, that it requires the TSize parameter even though it should not be necessary.
+        //-----------------------------------------------------------------------------
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TDim,
+            typename TSize,
+            template<std::size_t> class TTFnObj,
+            typename... TArgs>
+        ALPAKA_FN_HOST_ACC auto createVecFromIndexedFn2(
```

Comment:
For conforming compilers there are already the versions without the `2` at the end (`createVecFromIndexedFn`, ...). Non conformant compilers have a workaround where those functions are within the class (which is not the purpose of a factory function).
Previously at every place where one of the factory functions was called, the correct one was selected manually via `#ifdef`.
Those new functions (`createVecFromIndexedFn2`, ...) abstract this manual selection away by providing an out of class version which calls the correct version (out-of-class/ in-class) for the underlying compiler.
Those functions are only a temporary hack and should be removed as soon as all compilers support the native out-of-class version.
Maybe we should call them `createVecFromIndexedFnWorkaround` or `createVecFromIndexedFnHack`?